source/Engine.cpp

```diff
@@ -2048,11 +2063,11 @@ void Engine::DoWeather(Weather &weather)
 		// Get all ship bodies that are touching a ring defined by the hazard's min
 		// and max ranges at the hazard's origin. Any ship touching this ring takes
 		// hazard damage.
-		for(Body *body : shipCollisions.Ring(Point(), hazard->MinRange(), hazard->MaxRange()))
+		for(Body *body : shipCollisions.Ring(weather.Origin(), hazard->MinRange(), hazard->MaxRange()))
 		{
 			Ship *hit = reinterpret_cast<Ship *>(body);
-			double distanceTraveled = hit->Position().Length() - hit->GetMask().Radius();
-			hit->TakeDamage(visuals, *hazard, multiplier, distanceTraveled, Point(), nullptr, hazard->BlastRadius() > 0.);
+			double distanceTraveled = (weather.Origin() - hit->Position()).Length() - hit->GetMask().Radius();
```

Comment:
This is functionally the same but makes use of existing Point methods for better readbility.
```suggestion
			double distanceTraveled = weather.Origin().Distance(hit->Position()) - hit->GetMask().Radius();
```